plat/kvm/arm/lcpu_start.S

```diff
@@ -0,0 +1,80 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Authors: Răzvan Vîrtan <virtanrazvan@gmail.com>
+ *          Justin He     <justin.he@arm.com>
+ *
+ * Copyright (c) 2021, Arm Ltd. All rights reserved.
+ * Copyright (c) 2021, University Politehnica of Bucharest. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+ #include <uk/asm.h>
+ #include <kvm-arm/smp_defs.h>
+
+.equ ENTRY_OFFSET, 8
+.equ STACK_OFFSET, 16
+.equ STATE_OFFSET, 40
+
+.text
+/* void _lcpu_start(arm64_cpu *cpu) */
+ENTRY(_lcpu_start)
+	/* save x0 */
+	mov x9, x0
+
+	/* Disable interrupts */
+	msr	daifset, #2
+
+	/* Set the context id */
+	msr	contextidr_el1, xzr
```

Comment:
According to the ARM documentation, the PROCID field (bits 31:0 of contextidr_el1) resets to an unknown value at a warm reset. This line brings a default initial value in `contextidr_el1`.

Reply:
Sure, but AFAIK we don't use `contextidr` anywhere, do we? There are many other registers that reset to an unknown value.

Reply:
Indeed, we are not using it, but it is also part of the initialization done in `plat/kvm/arm/entry64.S`. Since we may want to merge the AP entry code with the bootstrapping core one, as Marc suggested, I will keep it as it is for the moment and eventually we can remove it in the final code.